tempesta_fw/t/functional/very_many_backends/test_deadtime_1M.py

```diff
@@ -0,0 +1,333 @@
+        server = self.servers[1]
+        for listener in server.config.listeners:
+            sg.add_server(server.ip, listener.port, server.conns_n)
+        self.tempesta.config.add_sg(sg)
+
+    def append_extra_server_groups(self):
+        sgid = 0
+        for ifc in range(self.num_extra_interfaces):
+            server = self.servers[self.extra_servers_base + ifc]
+            for listener in server.config.listeners:
+                sg = tempesta.ServerGroup('extra-%i' % sgid)
+                sg.add_server(server.ip, listener.port, server.conns_n)
+                self.tempesta.config.add_sg(sg)
+                sgid += 1
+
+    def create_clients(self):
+        """ Override to set desired list of benchmarks and their options. """
+        self.client = DeadtimeClient(timeout=-1)
+
+    def setup_nginx_config(self, config):
+        config.enable_multi_accept()
+        config.set_worker_connections(32768)
+        config.set_workers(4096)
+        config.set_worker_rlimit_nofile(16384)
+        config.set_ka(timeout = 180)
+        for listener in config.listeners:
+            listener.backlog = 9000
+        config.build_config()
+        return
+
+    def create_servers(self):
+        self.servers = []
+        # default server
+        defport=tempesta.upstream_port_start_from()
+        server = control.Nginx(listen_port=defport)
+        self.setup_nginx_config(server.config)
+        self.servers.append(server)
+
+        server = control.Nginx(listen_port=self.base_port,
+                               ports_n=self.num_attempts,
+                               listen_ip=self.ips[0])
+        self.setup_nginx_config(server.config)
+        self.servers.append(server)
+
+        self.extra_servers_base = len(self.servers)
+        for ifc in range(self.num_extra_interfaces):
+            server = control.Nginx(listen_port=self.base_port,
+                                   ports_n=self.num_extra_ports,
+                                   listen_ip=self.ips[ifc + 1])
+            self.setup_nginx_config(server.config)
+            self.servers.append(server)
+
+    def pre_test(self):
+        remote.server.run_cmd("sysctl -w net.core.somaxconn=8192")
+        remote.server.run_cmd("sysctl -w net.ipv4.tcp_max_orphans=1000000")
+        remote.tempesta.run_cmd("sysctl -w net.core.somaxconn=8192")
+        remote.tempesta.run_cmd("sysctl -w net.ipv4.tcp_max_orphans=1000000")
+        remote.client.run_cmd("sysctl -w net.core.somaxconn=8192")
+        remote.client.run_cmd("sysctl -w net.ipv4.tcp_max_orphans=1000000")
+
+        for server in self.servers:
+            server.start()
+
+        self.tempesta.config.set_defconfig(self.config)
+        self.configure_tempesta()
+        self.tempesta.start()
+        self.client.start()
+
+    def post_test(self):
+        self.client.stop()
+        self.assert_clients()
+
+    def assert_clients(self):
+        assert self.client.long_times == 0, 'Too long deadtime'
```

Comment:
Too cryptic warning. better to use `Reconfiguration took too long time: server group was unavailable more than 1s.` Please make assertion warnings more readable for those who run the test but don't aware of it's internals.

Also `self.assertEqual()` has a more fancy output. It will also print value of `self.client.long_times`. Try, you should like it.

Reply:
fixed